elf2uf2/elf2uf2.cpp

```diff
@@ -11,49 +11,93 @@
 #include <cstring>
 #include <cstdarg>
 #include <algorithm>
-#include <cstring>
 #include <memory>
 
 #include "elf2uf2.h"
 #include "errors.h"
 
+// Size of a flash sector in bytes
 #define FLASH_SECTOR_ERASE_SIZE 4096u
 
+// Define the page size for UF2 (256 Bytes)
+// These definitions are already present in the header file and should not be redefined here.
+// #define LOG2_PAGE_SIZE 8u
+// #define UF2_PAGE_SIZE (1u << LOG2_PAGE_SIZE)
+
+// Global variable to control verbose mode
 static bool verbose;
 
+/**
+ * @brief Terminates the program with a read error.
+ */
 static void fail_read_error() {
     fail(ERROR_READ_FAILED, "Failed to read input file");
 }
 
+/**
+ * @brief Terminates the program with a write error.
+ */
 static void fail_write_error() {
     fail(ERROR_WRITE_FAILED, "Failed to write output file");
 }
 
+/**
+ * @struct page_fragment
+ * @brief Represents a fragment of a page with file offset and size.
+ */
 struct page_fragment {
-    page_fragment(uint32_t file_offset, uint32_t page_offset, uint32_t bytes) : file_offset(file_offset), page_offset(page_offset), bytes(bytes) {}
-    uint32_t file_offset;
-    uint32_t page_offset;
-    uint32_t bytes;
+    /**
+     * @brief Constructor for page_fragment.
+     * @param file_offset Offset in the input file.
+     * @param page_offset Offset within the page.
+     * @param bytes Number of bytes in the fragment.
+     */
+    page_fragment(uint32_t file_offset, uint32_t page_offset, uint32_t bytes)
+        : file_offset(file_offset), page_offset(page_offset), bytes(bytes) {}
+
+    uint32_t file_offset;  ///< Offset in the input file.
+    uint32_t page_offset;  ///< Offset within the page.
+    uint32_t bytes;        ///< Number of bytes in the fragment.
 };
 
+/**
+ * @brief Checks if an address is within the valid ranges.
```

Comment:
It checks both `addr` and `addr + size`, so perhaps this should say "Checks if a region is within the valid ranges." ?